lib/api/services/wallet_service.dart

```diff
@@ -243,68 +183,152 @@ class WalletService extends ApiService {
         }),
       );
 
-
-
       if (transactionsData.statusCode == 200) {
         // If the server did return a 200 OK response,
         // then parse the JSON.
         TransactionListResponse transactionListResponse =
-        TransactionListResponse.fromJson(jsonDecode(transactionsData.body));
-
-
-          List<Transaction> transactionsList = [];
-          for (TransactionGroupResponse transactions in transactionListResponse.transactions!) {
-
-            for(TransactionsResponse t in transactions.transactions!){
-              Transaction transaction = Transaction(
-                  type: transactions.type!,
-                  timestamp: transactions.timestamp!,
-                  block: transactions.block!,
-                  address: ((t.sender != null) ? t.sender! : t.destination!),
-                  amount: t.amount, fee:transactions.fee!,
-                  baseAddress: walletAddress
-              );
-              transactionsList.add(transaction);
-            }
+            TransactionListResponse.fromJson(jsonDecode(transactionsData.body));
+
+        List<Transaction> transactionsList = [];
+        for (TransactionGroupResponse transactions
+            in transactionListResponse.transactions) {
+          for (TransactionsResponse t in transactions.transactions) {
+            Transaction transaction = Transaction(
+                type: transactions.type,
+                timestamp: transactions.timestamp,
+                block: transactions.block,
+                address: ((t.sender != null) ? t.sender! : t.destination!),
+                amount: t.amount,
+                fee: transactions.fee,
+                baseAddress: walletAddress);
+            transactionsList.add(transaction);
           }
-          return TransactionsGroup(address: walletAddress, transactionsList: transactionsList);
-
+        }
+        return TransactionsGroup(
+            address: walletAddress, transactionsList: transactionsList);
       } else {
-
-        throw Exception('${transactionsData.body}');
+        throw Exception(transactionsData.body);
       }
     } on Exception catch (e) {
       throw Exception(e.toString());
     }
   }
 
-
-
-  // @POST("/v1/send")
   Future<dynamic> sendXCH(
       {required String privateKey,
-      required var amount,
-      required String address,required int fee}) async {
+      required String address,
+      required int amount,
+      required int fee}) async {
+    var totalAmount = amount + fee;
+    var privateKeyObject = PrivateKey.fromBytes(
```

Comment:
Would be nice: Lines 223-228 appear to repeat in a few places in this changeset thus could be put into a separate function and called

Reply:
Not sure how this would be abstracted easily. Any recommendations?